### PR TITLE
NXP-30589: log warn instead of throwing an exception on unallowed ass…

### PR DESCRIPTION
**Pull request by guirenard**
…ignement

### CODE REVIEW

File: addons/nuxeo-platform-document-routing/nuxeo-routing-rest-api/src/test/java/org/nuxeo/ecm/restapi/server/WorkflowEndpointTest.java [hunk 1]
```diff
@@ -512,13 +512,17 @@ public void testSecurityCheckOnGlobalVariable() throws IOException {
...
+            assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+        }
+
+        try (CloseableClientResponse response = getResponse(RequestType.GET,
+                "/workflow/" + createdWorkflowInstanceId)) {
+            assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+            JsonNode workflowInstance = mapper.readTree(response.getEntityInputStream());
+            JsonNode variables = workflowInstance.get("variables");
+            // but the global variable has not been modified
+            assertTrue(variables.get("review_result").isNull());
```
Comment (NourNuxeo) [inline addons/nuxeo-platform-document-routing/nuxeo-routing-rest-api/src/test/java/org/nuxeo/ecm/restapi/server/WorkflowEndpointTest.java:536]: maybe assert the log was created ?

Comment (guirenard) [reply addons/nuxeo-platform-document-routing/nuxeo-routing-rest-api/src/test/java/org/nuxeo/ecm/restapi/server/WorkflowEndpointTest.java:536]: I did not think about it. Do you have any pointer where we already do that?

Comment (NourNuxeo) [reply addons/nuxeo-platform-document-routing/nuxeo-routing-rest-api/src/test/java/org/nuxeo/ecm/restapi/server/WorkflowEndpointTest.java:536]: https://github.com/nuxeo/nuxeo-lts/blob/d32456e02f736179cc82d281ac07bd4bdd1b2034/modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryQuery.java#L3608

The LogCaptureFeature may be what you ask about ?

**Review by ataillefer — COMMENTED** [review]
:+1: apart from Nour's comment.

File: addons/nuxeo-platform-document-routing/nuxeo-routing-core/src/test/java/org/nuxeo/ecm/platform/routing/test/GraphRouteTest.java [hunk 2]
```diff
@@ -46,6 +48,7 @@
 import org.nuxeo.ecm.automation.OperationChain;
 import org.nuxeo.ecm.automation.OperationContext;
 import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.test.AutomationFeature;
```
Comment (ataillefer) [inline addons/nuxeo-platform-document-routing/nuxeo-routing-core/src/test/java/org/nuxeo/ecm/platform/routing/test/GraphRouteTest.java]: Unneeded?

**Review by ataillefer — DISMISSED** [review]

**Review by NourNuxeo — DISMISSED** [review]

**Comment by guirenard** [comment]
Rebased on 10.10 branch

**Review by troger — APPROVED** [review]

**Review by NourNuxeo — APPROVED** [review]

**Review by ataillefer — APPROVED** [review]

**Pull request merged**